packages/react-vapor/src/components/textarea/TextArea.tsx

```diff
@@ -57,43 +57,61 @@ const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITextAreaOwnProps): I
     onUnmount: () => dispatch(removeTextArea(ownProps.id)),
 });
 
-export class TextArea extends React.Component<ITextAreaProps, {}> {
-    static defaultProps: Partial<ITextAreaOwnProps> = {
-        additionalAttributes: {},
-        className: '',
+export const TextArea: React.FunctionComponent<ITextAreaProps> = (props) => {
+    const [debouncedValue, setDebouncedValue] = React.useState(props.value);
+    const [isValid, setIsValid] = React.useState(true);
+
+    React.useEffect(() => {
+        setTimeout(() => {
+            setDebouncedValue(props.value);
+        }, 300);
+    }, [props.value]);

```

Comment:
Here you will create a lot of timeout without cancelling them if the value changes, your effect should return the cancelling function:

```typescript
React.useEffect(() => {
        const timeout = setTimeout(() => setDebouncedValue(props.value), 300);
        return () => clearTimeout(timeout);
    }, [props.value]);
```